bootable-jar/WFLY-14934_Server_Mode.adoc

```diff
@@ -0,0 +1,123 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14934[WFLY-14934]
+
+=== Related Issues
+
+* New WildFly S2I: https://issues.redhat.com/browse/WFLY-14936[WFLY-14936]
+
+* JKube evolved support: https://issues.redhat.com/browse/WFLY-14936[WFLY-14938]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+* https://github.com/eclipse/jkube[JKube]
+
+* https://github.com/wildfly/wildfly-s2i[WildFly s2i]
+
+* https://github.com/wildfly/quickstart[WildFly quickstarts]
+
+* ODO dev files.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+Ability to build and run an un-packaged server. All the goals defined in the Maven plugin are applicable to a server produced inside an installation dir.
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* Plugin behave the same as when producing a Bootable JAR, the only visible difference is that the project target dir contains 
+   a directory (named server by default) that contains the server and any deployments.
+* The Server mode is activated by the ```<server>``` configuration element. 
+* The server mode can be enabled/disabled, e.g.: ```<server> <enabled>[true|false]</enabled></server>```
+* In server mode, multiple deployments can be deployed inside the server. The primary one (the current project if producing a primary artifact) 
+   but also any war/ear files added as Maven dependencies in the project pom.xml file.
+* In bootable JAR mode, a single deployment can be added (the primary or dependency one).
+* By default the server is generated in ```target/server``` directory. The plugin offers a configuration item to override the name of the directory generated in the
+   ```target``` directory. E.g.: ```<server> <directory-name>foo-server</directory-name></server>```
+* In server mode, in order to allow for execution of a launch script for start/run/dev/dev-watch modes in a cloud 
+   context (eg: ODO), a launcher bash script file name (located in $JBOSS_HOME/bin) can be provided. For example, 
+   when using the legacy cloud galleon feature-pack, one can provide the ``` openshift-launch.sh``` script file name that this galleon feature-pack provisions.
+   For example: ```<server> <launch-script>openshift-launch.sh</launch-script></server>```
```

Comment:
again, this script is relative to `$JBOSS_HOME/bin`.
Can I pass an absolute path instead? `<server> <launch-script>/my/own/openshift-launch.sh</launch-script></server>`

Reply:
The launch script should be relative to the bin directory. I didn't open the door to absolute script path.